frontends/concrete-python/concrete/fhe/compilation/server.py

```diff
@@ -88,6 +94,17 @@ def __init__(
         self._server_program = server_program
         self._mlir = None
         self._composition_rules = composition_rules
+        self._client_to_prepare_clear_values = None
+
+        functions_parameters = json.loads(client_specs.client_parameters.serialize())["circuits"]
```

Comment:
You can't do that just for main, especially it will not works with modules

Reply:
Fixed!